backend/inspirehep/records/views.py

```diff
@@ -82,6 +86,76 @@ def get(self, pid, record):
         return jsonify(data)
 
 
+class WorkflowsRecordSourcesResource(MethodView):
+
+    view_name = "workflows_record_sources"
+    decorators = [
+        login_required_with_roles([Roles.superuser.value, Roles.cataloger.value])
+    ]
+
+    def get(self):
+        record_uuid = request.json.get("record_uuid")
+        if not record_uuid:
+            return jsonify({"message": "record_uuid must be passed"}), 500
+        query = WorkflowsRecordSources.query.filter_by(record_uuid=str(record_uuid))
+        source = request.json.get("source")
+        if source:
+            query = query.filter_by(source=source.lower())
+        results = query.all()
+        if not results:
+            return jsonify({"message": "Workflow source not found"}), 404
+        results_data = [
+            {
+                key: val
+                for key, val in result.__dict__.items()
+                if key in ["created", "json", "record_uuid", "source", "updated"]
+            }
+            for result in results
+        ]
+        return jsonify({"workflow_sources": results_data}), 200
+
+    def post(self):
+        record_uuid = request.json.get("record_uuid")
+        source = request.json.get("source")
+        root_json = request.json.get("root_json")
+        if not all([record_uuid, source, root_json]):
+            return jsonify({"message": "all the arguments must be passed"}), 500
```

Comment:
can we just use `webargs`?

Reply:
sure, I wanted to be consistent with other views ;)